blockchain/chain/src/store/chain_store.rs

```diff
@@ -423,6 +437,78 @@ where
     })
 }
 
+//this attempts to deserialize to unsigend message or signed message and then returns it at as a message trait object
+pub fn get_chain_message<DB>(db: &DB, key: &Cid) -> Result<BoxMessage, Error>
+where
+    DB: BlockStore,
+{
+    let value = db
+        .read(key.key())?
+        .ok_or_else(|| Error::UndefinedKey(key.to_string()))?;
+    if let Ok(message) = from_slice::<UnsignedMessage>(&value) {
+        Ok(Box::new(message))
+    } else {
+        let signed_message: SignedMessage = from_slice(&value)?;
+        Ok(Box::new(signed_message))
+    }
+}
+
+//given a tipset this functtion will return all messages as a trait object
+pub fn messages_for_tipset<DB>(db: &DB, ts: &Tipset) -> Result<Vec<BoxMessage>, Error>
+where
+    DB: BlockStore,
+{
+    let mut applied: HashMap<Address, u64> = HashMap::new();
+    let mut balances: HashMap<Address, BigUint> = HashMap::new();
+    let state = StateTree::new_from_root(db, ts.parent_state())?;
+
+    //message to get all messages for block_header into a single iterator
```

Comment:
idk small nit but for comments can you add a space after the `//` for consistency?